### DISCUSSION TITLE
Maximum rows in an Excel file that Import-Excel can read

### DISCUSSION THREAD

**Discussion by jharrod3003** [dfinke/ImportExcel#1405] [Q&A]
Hi all,

I'm writing a PowerShell script to read an Excel file containing the maximum row size (1,048,576) and 69 columns.
However, I am getting an error message as below.

**MethodInvocationException: < Import-Excel install path >\Import-Excel.ps1:124**
`Line 124: else { $ExcelPackage.Load($stream) }`
**Exception calling "Load" with 1 arguments(s). Array dimensions exceed supported range.**
**WARNING: Selecting columns 1 to 0 might give odd results** - I'm not passing any columns to it?

I have used the module previously to read other Excel files, but none over 50,000 rows until this file.

Any help will be appreciated.

Thanks!
James Harrod

**Comment by dfinke** [comment]
Never tried with more than 100K rows. Could be out of memory. It is using a C# construct underneath to read the data, and it may have limitations.

**Reply by jharrod3003** [reply]
I forgot to mention that I'm working with version 7.4.2 and not the latest version. That is because I looked through the discussions before when it was really slow, and found that version 7.8.4 includes many enhancements which potentially make it slower with importing Excel files.
I also did some testing on rowcount and found that the maximum rows it can read is 901,750, and it took 7 minutes to read. I'm not sure if that will help anyone else, but thought it would be useful to know.

**Reply by dfinke** [reply]
Thanks for running and reporting that. Yes, 7.4.2 is the faster version. 

Wondering of the C# Load function may be setting an array limit.
